modAionImpl/src/org/aion/zero/impl/cli/Cli.java

```diff
@@ -52,496 +57,511 @@
-                                    System.out.println("Please choose valid directory name!\n");
-                                    return 1;
-                                }
-
-                            default:
-                                System.out.println("\nInvalid network selected!\n");
-                                System.out.println("--- Available Networks ---");
-                                System.out.println("    mainnet, conquest");
-                                System.out.println("--------------------------\n");
-                                return 1;
-                        }
+                }
+                System.out.println(Version.KERNEL_VERSION);
+                return EXIT;
+            }
 
-                    } else {
-                        System.out.println("\nInvalid network selected!");
-                        System.out.println("--- Available Networks ---");
-                        System.out.println("    mainnet , conquest");
-                        System.out.println("--------------------------\n");
-                        return 1;
-                    }
+            // 2. determine the network configuration
 
-                // Determines database folder path
-                case "-d":
-                case "--datadir":
-                    if ( (args.length == 2 || args.length == 4) && isValid(args[1]))  {
+            if (options.getNetwork() != null
+                    || (options.getConfig() != null && !options.getConfig().isEmpty())) {
+                String strNet = options.getNetwork();
+                // the network given in config overwrites the -n option
+                if (options.getConfig() != null && !options.getConfig().isEmpty()) {
+                    strNet = options.getConfig();
+                }
+                setNetwork(strNet, cfg);
+                // no return -> allow for other parameters combined with -n
+            }
 
-                        // -d [directory]
-                        if (args.length == 2) {
+            // 3. determine the execution folder path; influenced by --network
 
-                            copyNetwork(path + "/" + args[1], net);
-                            cfg.getLog().setLogPath(args[1] + "/" + net + "/log");
-                            cfg.getDb().setDatabasePath(args[1] + "/" + net + "/database");
-                            Keystore.setKeystorePath(path + "/" + args[1] + "/" + net + "/keystore");
-                            return 2;
+            if (options.getDirectory() != null) {
+                if (!setDirectory(options.getDirectory(), cfg)) {
+                    return ERROR;
+                }
+                // no return -> allow for other parameters combined with -d
+            }
 
-                        }
+            // reading from correct config file
+            File configFile = cfg.getExecConfigFile();
+            if (!configFile.exists()) {
```

Comment:
if the given config file doesn't exist, can we either:

1) exit with an error
2) use the default but also print out a warning that the given input was not valid so we used the initial config (also, is it possible that the default config doesn't exist either?)

I feel like the user could get confused is he/she inputs something invalid and the program silently ignores it and does its own thing

Reply:
I will explain the reasoning behind that code below.
First, keep in mind the following 3 points:
1. there exists a **base directory** where all the code and the `aion.sh` reside;
2. we allow setting an **execution path** with the `--datadir` option and/or the `--network` option;
3. we must maintain compatibility with old kernel executables.
So, in order to make these features compatible, I've opted for the following:
a) If I find a `config.xml` and `genesis.json` at the old location, I will attempt to execute the CLI command or run the kernel as previous kernels would, in the **base directory**.
b) But, if the user calls the kernel with either the `--datadir value`, `--network value` or `--config value` option, then it will take precedence over old executable style and behave in the following way:
-> we create an **execution path** which will be either a directory in the base path with the network name (to ensure that if I run different networks from the same base path, they will not clash with eachother) or the directory given in the datadir option plus the network;
-> we copy the configuration files to the **execution path** to save the configuration;
-> we create the `keystore`, `log` and `database` directories, if they do not alreasy exist.

Reply:
Because of this setup, there are in fact 3 places from where a kernel can read it's configuration:
1. `config/config.xml` like old kernels
2. `config/network/config.xml` for new kernels with new execution paths
3. `network/config/config.xml` for new kernels where the execution path already exists

Reply:
I agree that this will need to be well documented when we introduce the `--datadir` and `--network` options.